pkgs/servers/invidious/default.nix

```
@@ -0,0 +1,59 @@
+{ lib, crystal, fetchgit, git, libxml2, sqlite, openssl_1_0_2, yaml2json }:
+crystal.buildCrystalPackage rec {
+  pname = "invidious";
+  version = "0.19.1";
+
+  src = fetchgit {
+    url = "https://github.com/omarroth/invidious.git";
+    rev = version;
+    sha256 = "026ywx2a32m48lniwln5d54iscmvvc5q6y619aj5wvprwsyyqcz3";
```

I got a different hash, and so did ofborg
```suggestion
    sha256 = "1j8cpijgin8p1hhg89pcmls59675rs9vir45dycas30cl4xmh4cn";
```

---

Weird, I was pretty sure I checked for this. I hope the `deepClone = true` isn't the cause of this.

---

Oh yeah and just now when I wanted to build it with that hash:
```
hash mismatch in fixed-output derivation '/nix/store/zh66m0qy9rwlm9qkvvdl7q90qgn2nfqd-invidious':
  wanted: sha256:1j8cpijgin8p1hhg89pcmls59675rs9vir45dycas30cl4xmh4cn
  got:    sha256:026ywx2a32m48lniwln5d54iscmvvc5q6y619aj5wvprwsyyqcz3
```

I think `deepClone` *is* the cause of it unfortunately..

---

Very weird, and unfortunate :(

---

@jonringer Can you upload the result of this? That's the store path you're getting
```
nix-store --export /nix/store/x9ra6k5lg8g5wlpy3kr0vd81fwn5gkjx-invidious > exported
```

I'm also uploading the result of the one I'm getting:
```
nix-store --export /nix/store/zh66m0qy9rwlm9qkvvdl7q90qgn2nfqd-invidious > exported
```

Here's mine: https://send.firefox.com/download/8a55393099407fc4/#n84JTTDS9jRoVXxxrGhF5Q

I'd like to compare the two to see the differences

---

didn't feel like signing in... so it will expire after the first download :(

https://send.firefox.com/download/a6ba584f653e5b2d/#Z4UAYX83NB3lrLHKYcpiDQ

---

Nice thanks, I was able to find some non-determinism in `nix-prefetch-git` because of this. Fixed in https://github.com/NixOS/nixpkgs/pull/67805, which I also put in this PR. So if you try it now it should work :D

---

that's some good detective work, sir

---

you're going to hate me.... but I just got:
```
hash mismatch in fixed-output derivation '/nix/store/j02fmn53mbpgmw4ff1m430i62sspy80d-invidious':
  wanted: sha256:026ywx2a32m48lniwln5d54iscmvvc5q6y619aj5wvprwsyyqcz3
  got:    sha256:1z3s9qq6bzzgwjyijffy46cby2f8n03461a0ykpwsjhlyd9ls8bf
```

---

Aw jeez

---

Ah wait, I didn't push the hash update yet, that's indeed the hash I'm also getting :D

---

IIRC, there's ~3 different ways git can pack the history, and it was never meant to be deterministic in packing.  But this is outside my realm of comfort.

---

*Now* it should work

---

```
[1 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/67805
10 package were build:
bundix cabal2nix crystal2nix dep2nix go2nix nix-prefetch-git nix-prefetch-scripts nix-update-source pypi2nix vgo2nix
```
all green :)